### DISCUSSION TITLE
Nested not found pages

### DISCUSSION THREAD

**Discussion by darklight9811** [vercel/next.js#53486] [Ideas]
### Goals

1. allow the not found page to be scoped
2. makes next automatically trigger closest not found

### Background

This complicates the life of applications that uses localization. And it differs from the error boundary behavior, making this look like a bug. And it seems counter intuitive to trigger them only on manual calls.

When there are several people complaining that this looks like a bug, I really think this should be look into.

### Proposal

When there are sub not-found files in the app router directories, fallback to them instead of root one. This allows to fit them in layouts/templates/etc, enabling the full power of app router.

**Comment by Daniel4lm** [comment]
If I'm not mistaken, rendering local not-found.tsx pages was possible until ver. 13.4.9 https://github.com/vercel/next.js/issues/48227. I had to rollback from ver. 13.4.12 to 13.4.9 because of this.

**Reply by bregtemundo** [reply]
I can also confirm that rolling back to 13.4.9 fixed the not-found pages. 

with later versions of next I had a lot of issues.

For my case I had a app/[locale][[..slug]]/page.jsx -> that executes the notFound function when data was not loaded. 
middleware rewrote slugs to make sure the all go to locale folder.
Only when putting the not-found.tsx in app/not-found.tsx the custom error was shown.
layout.tsx from app/[locale]/.. was not on the not-found page making the styling to not be in line with the rest of the pages.
When build I got hydration errors, causing the 404 page to appear and then immediately disappear again. (This was not the case in dev mode)

**Comment by affanshahid** [comment]
It should be possible to define `not-found`s for nested dynamic routes. The [i18n example](https://github.com/vercel/next.js/tree/canary/examples/app-dir-i18n-routing) doesn't have a root layout, the "root" layout is defined inside the `[lang]` param hierarchy, so the default way of doing `not-found` wouldn't work with the official example. This doesn't make sense

**Comment by kavyantic** [comment]
+1

**Comment by wilstaley** [comment]
+1

The example in the docs for i18n shows the root layout file nested within a [lang] directory.  I would expect to also be able to have a not-found file at the same place.
